src/packages/recompose/isReferentiallyTransparentFunctionComponent.js

```diff
@@ -5,7 +5,7 @@ const isReferentiallyTransparentFunctionComponent = Component => Boolean(
   !isClassComponent(Component) &&
   !Component.defaultProps &&
   !Component.contextTypes &&
-  !Component.propTypes
+  (!Component.propTypes || process.env.NODE_ENV === 'production')
```

Comment:
swapping on `process.env.NODE_ENV === 'production' || !Component.propTypes` will avoid getting the `propTypes`